catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -848,4 +874,13 @@ private void restoreFromBackup(StageBackup backup) {
 			CameraManager.getInstance().resumeForScene();
 		}
 	}
+
+	private float calculateScreenRatio() {
+		DisplayMetrics metrics = Resources.getSystem().getDisplayMetrics();
+		XmlHeader header = ProjectManager.getInstance().getCurrentProject().getXmlHeader();
```

Comment:
We dig here very deep to find the XML header. I wonder if it's really the best solution if the project holds the XML. Is the architecture here ok?

Reply:
Device and project height/width are not the same thing. 
If a project is created on a device height and width of that device are saved in the project itself, therefore if you download it on another device later you have the same project (same aspect ratio). For that you can press in the pause menu on Stretch/Maximize to either stay at the aspect ratio of the creator device or use your whole screen. 
Since in the other stitching tickets this stuff is also pretty important I think it is completely intended to have that architecture like that ^^

Also there i've not found a better way to get a more accurate solution since this Stitching Feature should be pretty accurate over all devices (and also projects from other devices)